vpr/src/util/mmap_file.cpp

```diff
@@ -0,0 +1,35 @@
+#include "mmap_file.h"
+#include "vpr_error.h"
+#include "vtr_log.h"
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <unistd.h>
+
+#include "kj/filesystem.h"
+
+MmapFile::MmapFile(const std::string& file)
+    : size_(0) {
+    try {
+        auto fs = kj::newDiskFilesystem();
+        auto path = fs->getCurrentPath().evalNative(file);
+
+        const auto& dir = fs->getRoot();
+        auto stat = dir.lstat(path);
+        auto f = dir.openFile(path);
+        size_ = stat.size;
+        data_ = f->mmap(0, stat.size);
```

Comment:
Should there be an munmap in a destructor somewhere?

Reply:
Object is RAII, so munmap is done on destruction.